src/Microsoft.OData.Edm/Csdl/CsdlJsonWriter.cs

```diff
@@ -100,13 +150,43 @@ private void WriteSchemata()
             }
         }
 
+        /// <summary>
+        /// Schema Object asynchronously
+        /// </summary>
+        /// <returns></returns>
+        private async Task WriteSchemataAsync()

```

Comment:
```suggestion
        private async Task WriteSchemaAsync()

```

Reply:
@ElizabethOkerio This is done for both `WriteSchemata `and `WriteSchemataAsync`

> WriteSchemata to WriteSchema
> WriteSchemataAsync to WriteSchemaAsync

Reply:
Just wanted to point out that `Schemata` is the plural of `Schema`. Maybe this method was intentionally named so to indicate that it's meant to write more than one _schemas_. With that in mind, if we have a problem with the word "schemata", maybe we could rename the method to `WriteSchemas` instead.

Reply:
@ElizabethOkerio Please see the previous comment.
@habbes